controllers/operator.go

```diff
@@ -210,31 +219,52 @@ func (o *operator) newMetricsProvider() providers.MetricsProvider {
 }
 
 func (o *operator) syncDeploys(managedDeploys appsv1.DeploymentList) {
-	trackedPartitions := make(map[int32]bool)
+	// TODO: check that maximum partition is not greater than configured in the spec and ?warn?change?
+	// TODO: consider making `numPartitions` optional and just get it from the production rate metrics
```

Comment:
@hagen1778 wdyt about this?

Reply:
Partitions number change is pretty rare event and deserves manual intervention from my point of view.

Reply:
Partitions number change is pretty rare event and deserves manual intervention from my point of view.